Forge/src/main/java/me/ferdz/placeableitems/client/renderer/tileentity/FluidHolderRenderer.java

```diff
@@ -0,0 +1,184 @@
+
+            bufferPos(buffer, bounds.getBackBottomRight()).color(red, green, blue, alpha).tex(u2, v1).lightmap(southLMa, southLMb).endVertex();
+            bufferPos(buffer, bounds.getBackTopRight()).color(red, green, blue, alpha).tex(u2, v2).lightmap(southLMa, southLMb).endVertex();
+            bufferPos(buffer, bounds.getBackTopLeft()).color(red, green, blue, alpha).tex(u1, v2).lightmap(southLMa, southLMb).endVertex();
+            bufferPos(buffer, bounds.getBackBottomLeft()).color(red, green, blue, alpha).tex(u1, v1).lightmap(southLMa, southLMb).endVertex();
+        }
+
+        if (model.shouldRender(state, Direction.WEST)) {
+            int westCombined = getWorld().getCombinedLight(pos.west(), 0);
+            int westLMa = westCombined >> 16 & 65535;
+            int westLMb = westCombined & 65535;
+
+            bufferPos(buffer, bounds.getBackBottomLeft()).color(red, green, blue, alpha).tex(u1, v2).lightmap(westLMa, westLMb).endVertex();
+            bufferPos(buffer, bounds.getBackTopLeft()).color(red, green, blue, alpha).tex(u2, v2).lightmap(westLMa, westLMb).endVertex();
+            bufferPos(buffer, bounds.getFrontTopLeft()).color(red, green, blue, alpha).tex(u2, v1).lightmap(westLMa, westLMb).endVertex();
+            bufferPos(buffer, bounds.getFrontBottomLeft()).color(red, green, blue, alpha).tex(u1, v1).lightmap(westLMa, westLMb).endVertex();
+        }
+
+        if (model.shouldRender(state, Direction.EAST)) {
+            int eastCombined = getWorld().getCombinedLight(pos.east(), 0);
+            int eastLMa = eastCombined >> 16 & 65535;
+            int eastLMb = eastCombined & 65535;
+
+            bufferPos(buffer, bounds.getFrontBottomRight()).color(red, green, blue, alpha).tex(u1, v1).lightmap(eastLMa, eastLMb).endVertex();
+            bufferPos(buffer, bounds.getFrontTopRight()).color(red, green, blue, alpha).tex(u2, v1).lightmap(eastLMa, eastLMb).endVertex();
+            bufferPos(buffer, bounds.getBackTopRight()).color(red, green, blue, alpha).tex(u2, v2).lightmap(eastLMa, eastLMb).endVertex();
+            bufferPos(buffer, bounds.getBackBottomRight()).color(red, green, blue, alpha).tex(u1, v2).lightmap(eastLMa, eastLMb).endVertex();
+        }
+    }
+
+    /**
+     * Bind a {@link PlaceableItemsBlock} to a {@link FluidModel} implementation.
+     *
+     * @param block the block for which to bind the model
+     * @param model the model to bind
+     *
+     * @return this instance. Allows for chained method calls
+     */
+    public FluidHolderRenderer bind(PlaceableItemsBlock block, FluidModel model) {
+        this.fluidModels.put(block, model);
+        return this;
+    }
+
+    /**
+     * Clear all binded fluid models from this renderer. Ill-advised for modders to invoke this method.
+     * Unexpected issues may arise including the client crashing due to missing models.
+     */
+    public void clearFluidModels() {
+        this.fluidModels.clear();
+    }
```

Comment:
If this isn't needed & bad to use, why keep it?

Reply:
I believe I had this for the sake of client-sided cleanup but I'm not sure it's necessary. This can be removed!